### PR TITLE
Streamline technical overview docs

### PR DESCRIPTION
**Pull request by willingc**
This PR consolidates two technical documents (How It Works and Technical Overview) into one:
- update table of contents so there is only one Technical Overview section
- deduplicates content found in other sections
- tries to have the Technical Overview give readers a view of what components make up JupyterHub and how they intereact.

[Rendered Technical Overview doc page](http://test-jupyterhub.readthedocs.io/en/tech-overview/technical-overview.html)

screen shot 2017-07-24 at 10 25 45 pm

### CODE REVIEW

File: docs/source/technical-overview.md [hunk 1]
```diff
@@ -1,104 +1,134 @@
...
-  starts these servers is called a **Spawner**.
 - **Proxy**: the public facing part of JupyterHub that uses a dynamic proxy
   to route HTTP requests to the Hub and Single User Notebook Servers.
-- **Hub**: manages user accounts, authentication, and coordinates Single User
-  Notebook Servers using a Spawner.
+  [configurable http proxy](https://github.com/jupyterhub/configurable-http-proxy)
+  (node-http-proxy) is the default proxy.
 
-![JupyterHub subsystems](images/jhub-parts.png)
+- **Single-User Notebook Server** (Python/IPython/Tornado): a dedicated,
```
Comment (minrk) [inline docs/source/technical-overview.md]: IPython is not involved here. Probably just "Python/Tornado" or "Python/Jupyter Notebook/Tornado".

File: docs/source/technical-overview.md [hunk 2]
```diff
@@ -1,104 +1,134 @@
...
 the domain name of the server.
 
-Basic principles of operation:
+The basic principles of operation are:
 
-* Hub spawns proxy
-* Proxy forwards all requests to hub by default
-* Hub handles login, and spawns single-user servers on demand
-* Hub configures proxy to forward url prefixes to single-user servers
+- The Hub spawns proxy
```
Comment (minrk) [inline docs/source/technical-overview.md]: The Hub spawns the proxy in the default configuration, but is not a general principle of the subsystem interaction. Not sure if that should change here, but it's a bit imprecise, now.

File: docs/source/technical-overview.md [hunk 3]
```diff
@@ -1,104 +1,134 @@
...
+## The Process from JupyterHub Access to User Login
 
-See [Security documentation](#security) for how to configure JupyterHub to use SSL,
-or put it behind SSL termination in another proxy server, such as nginx.
+When a user accesses JupyterHub, the following events take place:
 
----
+- Login data is handed to the [Authenticator](./authenticators.html) instance for
+  validation
+- The Authenticator returns the username and if the login information is valid
```
Comment (minrk) [inline docs/source/technical-overview.md]: "returns the username <del>and</del> if the login information is valid"

File: docs/source/technical-overview.md [hunk 4]
```diff
@@ -1,104 +1,134 @@
...
+- A single-user notebook server instance is [spawned](./spawners.html) for the
+  logged-in user
+- When the single-user notebook server starts, the proxy is notified to forward
+  requests to `/user/[username]/*` to the single-user notebook server.
+- Two cookies are set, one for `/hub/` and another for `/user/[username]`,
```
Comment (minrk) [inline docs/source/technical-overview.md]: The `/user/` cookie is no more in 0.8, so only one cookie is set on /hub/

File: docs/source/technical-overview.md [hunk 5]
```diff
@@ -1,104 +1,134 @@
...
+- The browser is redirected to `/user/[username]`, and the request is handled by
+  the single-user notebook server.
 
-**Deprecation note:** Removed `--no-ssl` in version 0.7.
+Logging into a single-user notebook server is authenticated via the Hub:
 
-JupyterHub versions 0.5 and 0.6 require extra confirmation via `--no-ssl` to
-allow running without SSL using the command `jupyterhub --no-ssl`. The
-`--no-ssl` command line option is not needed anymore in version 0.7.
+- On request, the single-user server forwards the encrypted cookie to the Hub
```
Comment (minrk) [inline docs/source/technical-overview.md]: This sequence is now OAuth:

> the single-user server identifies the user with the Hub via OAuth
>
> - on request, the single-user server checks a cookie
> - if no cookie is set, redirect to the Hub for verification via OAuth
> - after verification at the Hub, the browser is redirected back to the single-user server
> - the token is verified and stored in a cookie
> - if no user is identified, the browser is redirected back to /hub/login

**Comment by willingc** [comment]
Thanks @minrk for updating for 0.8. I was hoping you would flag that in the review, and you did. Making changes and then I'll merge.

File: docs/source/technical-overview.md (hunk 2 above)
Comment (willingc) [inline docs/source/technical-overview.md]: I'll clarify for default config.

**Pull request merged**